### ISSUE TITLE
Installation issue

### ISSUE THREAD

**Issue by tolew1** [thorsten/phpMyFAQ#2960]
I'm using version 3.2.6 trying to do a fresh install.  PHP version 8.3.6,  AlmaLinux release 9.3

I keep getting an error message 

```
Gateway Timeout
The gateway did not receive a timely response from the upstream server or application.
```

This is in the logs.

```
[21-May-2024 09:01:53 Europe/Berlin] phpMyFAQ TypeError: phpMyFAQ\Language::setLanguage(): Argument #1 ($configDetection) must be of type bool, null given, called in /var/www/help.example.tv/html/index.php on line 76 in /var/www/help.example.tv/html/src/phpMyFAQ/Language.php on line 94
Stack trace:
#0 /var/www/help.example.tv/html/index.php(76): phpMyFAQ\Language->setLanguage()
#1 {main}
```

**Comment by modelrailroader** [comment]
I can't reproduce this. Do you have access to the database? Could you take a look at the table `faqconfig` if all the config entries are installed correctly?

**Comment by tolew1** [comment]
Weird. this time even though it still timed out after a good while, I went to the main URL and it pulled up the site. Admin seems to be working also. I don't know if anything is missing or if it actually finished the install or not. 

However updating the site config is super slow.  The frontend doesn't appear to navigate slowly so far but I also haven't added any faqs. You can see the load time here.

![image](https://github.com/thorsten/phpMyFAQ/assets/3882817/0b80edb6-474c-4df3-846e-ef52849331b6)

**Comment by modelrailroader** [comment]
That's absolutely not normal. But there are config items in the table named above?

**Comment by tolew1** [comment]
Yes I see the table is full. It's so weird. I've done several of these installs before and never experienced this. I don't know what to check.

FYI, The DB is on its own server, and it uses backend IP and not public IP so it shouldn't be a speed or latency issue.  I'm using PostgreSQL 16. I even recreated the DB. 

Let me know if you see anything wrong with this output. I changed my URL for privacy. 


```
select * from amffaqconfig;
                 config_name                 |                      config_value
---------------------------------------------+---------------------------------------------------------
 main.currentVersion                         | 3.2.6
 main.currentApiVersion                      | 2.2
 main.language                               | language_en.php
 main.languageDetection                      | true
 main.phpMyFAQToken                          | b5b12dab2ef207c8efe1435fe0a76dc9
 main.administrationMail                     | info@example.com
 main.contactInformation                     |
 main.enableAdminLog                         | true
 main.enableRewriteRules                     | false
 main.enableUserTracking                     | true
 main.metaDescription                        | phpMyFAQ should be the answer for all questions in life
 main.metaKeywords                           |
 main.metaPublisher                          | Example
 main.send2friendText                        |
 main.titleFAQ                               | phpMyFAQ Codename Pontus
 main.urlValidateInterval                    | 86400
 main.enableWysiwygEditor                    | true
 main.enableWysiwygEditorFrontend            | false
 main.enableMarkdownEditor                   | false
 main.templateSet                            | default
 main.optionalMailAddress                    | false
 main.dateFormat                             | Y-m-d H:i
 main.maintenanceMode                        | false
 main.enableGravatarSupport                  | false
 main.enableGzipCompression                  | true
 main.customPdfHeader                        |
 main.customPdfFooter                        |
 main.enableSmartAnswering                   | true
 main.enableCategoryRestrictions             | true
 main.enableSendToFriend                     | true
 main.privacyURL                             |
 main.enableAutoUpdateHint                   | true
 security.loginWithEmailAddress              | false
 main.enableAskQuestions                     | false
 main.enableNotifications                    | false
 main.contactInformationHTML                 | false
 records.numberOfRecordsPerPage              | 10
 records.numberOfShownNewsEntries            | 3
 records.defaultActivation                   | false
 records.defaultAllowComments                | false
 records.enableVisibilityQuestions           | false
 records.numberOfRelatedArticles             | 5
 records.orderby                             | id
 records.sortby                              | DESC
 records.orderingPopularFaqs                 | visits
 records.disableAttachments                  | true
 records.maxAttachmentSize                   | 100000
 records.attachmentsPath                     | attachments
 records.attachmentsStorageType              | 0
 records.enableAttachmentEncryption          | false
 records.defaultAttachmentEncKey             |
 records.enableCloseQuestion                 | false
 records.enableDeleteQuestion                | false
 records.randomSort                          | false
 records.allowCommentsForGuests              | true
 records.allowQuestionsForGuests             | true
 records.allowNewFaqsForGuests               | true
 records.hideEmptyCategories                 | false
 records.allowDownloadsForGuests             | false
 records.numberMaxStoredRevisions            | 10
 records.enableAutoRevisions                 | false
 search.numberSearchTerms                    | 10
 search.relevance                            | thema,content,keywords
 search.enableRelevance                      | false
 search.enableHighlighting                   | true
 search.searchForSolutionId                  | true
 search.enableElasticsearch                  | false
 security.permLevel                          | basic
 security.ipCheck                            | false
 security.enableLoginOnly                    | false
 security.bannedIPs                          |
 security.ssoSupport                         | false
 security.ssoLogoutRedirect                  |
 security.useSslForLogins                    | false
 security.useSslOnly                         | false
 security.forcePasswordUpdate                | false
 security.enableRegistration                 | true
 security.domainWhiteListForRegistrations    |
 security.enableSignInWithMicrosoft          | false
 security.enableGoogleReCaptchaV2            | false
 security.googleReCaptchaV2SiteKey           |
 security.googleReCaptchaV2SecretKey         |
 spam.checkBannedWords                       | true
 spam.enableCaptchaCode                      | true
 spam.enableSafeEmail                        | true
 spam.manualActivation                       | true
 spam.mailAddressInExport                    | true
 socialnetworks.enableTwitterSupport         | false
 socialnetworks.twitterConsumerKey           |
 socialnetworks.twitterConsumerSecret        |
 socialnetworks.twitterAccessTokenKey        |
 socialnetworks.twitterAccessTokenSecret     |
 socialnetworks.disableAll                   | false
 seo.metaTagsHome                            | index, follow
 seo.metaTagsFaqs                            | index, follow
 seo.metaTagsCategories                      | index, follow
 seo.metaTagsPages                           | index, follow
 seo.metaTagsAdmin                           | noindex, nofollow
 seo.enableXMLSitemap                        | true
 mail.remoteSMTP                             | false
 mail.remoteSMTPServer                       |
 mail.remoteSMTPUsername                     |
 mail.remoteSMTPPassword                     |
 mail.remoteSMTPPort                         | 25
 mail.remoteSMTPEncryption                   |
 mail.remoteSMTPDisableTLSPeerVerification   | false
 ldap.ldapSupport                            | false
 ldap.ldap_mapping.name                      | cn
 ldap.ldap_mapping.username                  | samAccountName
 ldap.ldap_mapping.mail                      | mail
 ldap.ldap_mapping.memberOf                  |
 ldap.ldap_use_domain_prefix                 | true
 ldap.ldap_options.LDAP_OPT_PROTOCOL_VERSION | 3
 ldap.ldap_options.LDAP_OPT_REFERRALS        | 0
 ldap.ldap_use_memberOf                      | false
 ldap.ldap_use_sasl                          | false
 ldap.ldap_use_multiple_servers              | false
 ldap.ldap_use_anonymous_login               | false
 ldap.ldap_use_dynamic_login                 | false
 ldap.ldap_dynamic_login_attribute           | uid
 api.enableAccess                            | true
 api.apiClientToken                          |
 main.referenceURL                           | https://help.example.com
 security.salt                               | 76ad663d98dd2ab8a008d4f1d4a8427d

```

**Comment by modelrailroader** [comment]
Looks like it has to look. But I'm happy to see that its's working now. Please continue testing as well with adding custom content like faqs.
Maybe @thorsten will have to take a look when he returns from vacation.

**Comment by tolew1** [comment]
Yes it seems to not be working right. I tried to save another config and it took a bit again. Then it showed the login admin page in the middle of the config page.  So I clicked login and it then said "Dave, this conversation can serve no purpose anymore. Goodbye."

**Comment by tolew1** [comment]
So I tried many ways on one of my hosts (OVH).  I even recreated my DB vm and changed to public IP instead of private IP and same weird errors. 503 gateway timeouts and not completing install. The weird thing is I've set this up there a couple of times with no issue. I finally gave up and tried my other hosting provider and it worked as it used to. I'm thinking there is some serious networking issues that I can't see with their vrack feature. The weird thing is, only phpmyfaq was the only thing I've had a problem with there. I have over 10 vms and all my other sites work fine.  Anyway, I'll close for now since it's working elsewhere.
